frontend/components/pricing/PremiumAddonBox.vue

```diff
@@ -9,509 +9,351 @@ import {
+    / addon.extra_dashboard_effective_balance,
   )
 })
 
-const limitReached = computed(() => {
-  return quantityForPurchase.value >= maximumQuantity.value
-})
+const {
+  isStripeDisabled, stripeCustomerPortal, stripePurchase,
+} = useStripe()
+const { promoCode } = usePromoCode()
 
-const purchaseQuantityButtons = computed(() => {
-  return {
-    minus: {
-      callback: () => {
-        if (quantityForPurchase.value > 1) {
-          quantityForPurchase.value--
-        }
-      },
-      disabled: quantityForPurchase.value <= 1,
-    },
-    plus: {
-      callback: () => {
-        if (quantityForPurchase.value < maximumQuantity.value) {
-          quantityForPurchase.value++
-        }
-      },
-      disabled: limitReached.value,
-    },
+const isSubmitButtonDisabled = computed(() =>
+  isStripeDisabled.value
+  || purchaseQuantity.value > maximumPurchaseQuantityAllowed.value
+  || purchaseQuantity.value < 1,
+)
+
+const handleSubmitPurchase = async () => {
+  if (isStripeDisabled.value) {
+    return
   }
-})
+
+  if (isLoggedIn.value) {
+    if (addonSubscriptionCount.value > 0) {
+      await stripeCustomerPortal()
+    }
+    else {
+      await stripePurchase(
+        isPaymentYearly
+          ? addon.stripe_price_id_yearly
+          : addon.stripe_price_id_monthly,
+        purchaseQuantity.value,
+      )
+    }
+  }
+  else {
+    await navigateTo({
+      path: '/login', query: { promoCode },
+    })
+  }
+}
 </script>
 
 <template>
-  <div class="box-container">
-    <div class="summary-container">
-      <div class="validator-count">
-        {{ boxText.validatorCount }}
-        <div class="subtext">
-          {{ $t("pricing.addons.per_dashboard") }}
-          <BcTooltip
-            position="top"
-            :fit-content="true"
-          >
-            <FontAwesomeIcon
-              :icon="faInfoCircle"
-              class="tooltip-icon"
-            />
-            <template #tooltip>
-              <div class="saving-tooltip-container">
-                {{
-                  $t("pricing.pectra_tooltip", {
-                    effectiveBalance: formatNumber(
-                      props.addon?.extra_dashboard_validators * 32,
-                    ),
-                  })
-                }}
-              </div>
-            </template>
-          </BcTooltip>
-        </div>
-        <div class="per-validator">
-          {{ boxText.perValidator }}
-        </div>
+  <div class="premium-addon-box">
+    <div class="premium-addon-box__header">
+      <div>
+        <span class="premium-addon-box__title">
+          {{
+            $t('pricing.addons.effective_balance', {
+              amount: extraEffectiveBalance,
+            })
+          }}
+        </span>
+        <spans class="premium-addon-box__title-detail">
+          {{ $t('pricing.per_min_validator_deposit', {
+            amount: formatFiatCurrency(pricePerValidator, {
+              minimumFractionDigits: 4,
+            }),
```

Comment:
issue: we have two functions achieving the same thing

like this maybe: 
```suggestion
            amount: formatAmount(pricePerValidator, {
            sourceUnit: 'base',
            sourceCurrency: 'EUR',
            targetCurrency: 'EUR',
            minimumFractionDigits: 4,
            }),
```

Reply:
The two functions return different things (`XX EUR` vs `€XX`)
We may put them together in a separate ticket in the future.